.gitignore

```diff
@@ -0,0 +1 @@
+node_modules
```

Comment:
Why did you put everything into a single commit?

Reply:
Because I did it in a private repository, tried different ways to implement it, and at last copy the valid codes to this repository .